platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsAccessFromGroovyDslIntegrationTest.groovy

```diff
@@ -737,4 +738,38 @@ class IsolatedProjectsAccessFromGroovyDslIntegrationTest extends AbstractIsolate
             projectsConfigured(":", ":a", ":b")
         }
     }
+
+    def "fails on access #type of unconfigured project"() {
+        given:
+        settingsFile << """
+            include(':a')
+            include(':b')
+        """
+
+        file("a/build.gradle") << """
+            def unconfiguredProject = project(':b')
+            $referrerConfiguration
+        """
+
+        file("a/build.gradle") << """
+            import ${Property.name}
+
+            interface MyExtension {
+                Property<String> bar()
+            }
+
+            $referentConfiguration
+        """
+
+        when:
+        isolatedProjectsFails 'help', WARN_PROBLEMS_CLI_OPT
+
+        then:
+        failure.assertHasErrorOutput(expectedOutput)
```

Comment:
Should we also have a test that checks for an IP violation in such a setup?

Reply:
Ok